bciers/apps/reporting/src/middlewares/withRulesAppliedReport.ts

```diff
@@ -31,9 +56,52 @@ const handleIndustryUserRoutes = async (request: NextRequest, token: any) => {
       // 🛸 Redirect to BCIERS dashboard
       return NextResponse.redirect(new URL(`/onboarding`, request.url));
     }
+    const { pathname } = request.nextUrl;
+    const versionId = extractVersionId(pathname);
+
+    // Allow requests to "verification" if report needs verification
+    if (pathname.includes("verification")) {
+      const needVerification = await fetchApi(
+        `reporting/report-version/${versionId}/report-needs-verification`,
+        {
+          user_guid: token.user_guid,
+        },
+      );
+      if (!needVerification) {
+        // 🛸 Redirect to final review
+        return NextResponse.redirect(
+          new URL(`/reporting/reports/${versionId}/final-review`, request.url),
+        );
+      }
+    }
+
+    // Check if pathname contains any of the LFO operation type paths
+    if (reportPaths.some((path) => pathname.includes(path))) {
+      if (versionId) {
+        const reportOperation = await fetchApi(
+          `reporting/report-version/${versionId}/report-operation`,
+          {
+            user_guid: token.user_guid,
+          },
+        );
+        if (
+          reportOperation &&
+          reportOperation.report_operation.operation_type ===
+            "Single Facility Operation"
+        ) {
+          // 🛸 Redirect to review operator
+          return NextResponse.redirect(
+            new URL(
+              `/reporting/reports/${versionId}/review-operator-data`,
+              request.url,
+            ),
+          );
+        }
+      }
+    }
```

Comment:
This file is starting to get big enough that we would split the `if...then return , if...then return ...` into a stack of individual middlewares